src/perftest_resources.c

```diff
@@ -3689,7 +3693,7 @@ int run_iter_bw_server(struct pingpong_context *ctx, struct perftest_parameters
 					}
 					//coverity[uninit_use]
 					if ((user_param->test_type==DURATION || posted_per_qp[wc_id] + user_param->recv_post_list <= user_param->iters) &&
-							unused_recv_for_qp[wc_id] >= user_param->recv_post_list) {
+					    unused_recv_for_qp[wc_id] >= user_param->recv_post_list && !user_param->use_unsolicited_write) {
```

Comment:
Couldn't this additional condition cause performance issues / degradations ? especially when first condition isn't common

Reply:
We didn't see any performance effect in our testing. In particular, in the BW case there should always be enough posted requests.